tests/pkg_tweetnacl/main.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file
+ * @brief       tweetnacl NaCl crypto library example
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include <tweetnacl.h>
+#include "ps.h"
+
+static unsigned char alice_pk[crypto_box_PUBLICKEYBYTES];
+static unsigned char alice_sk[crypto_box_SECRETKEYBYTES];
+
+static unsigned char bob_pk[crypto_box_PUBLICKEYBYTES];
+static unsigned char bob_sk[crypto_box_SECRETKEYBYTES];
+
+static const char message[] = "0123456789abcdef";
+
+/* 17 == strlen(message) + 1 for null byte */
+#define MLEN (17U + crypto_box_ZEROBYTES)
```

Comment:
@kYc0o my proposal would make it less complex IMHO, especially considering maintainability. If someone changes `message`, they would also be needed to change this magic number which wouldn't be the case with my proposal ;-)